cdap-ui/app/cdap/components/PreviewData/Table.tsx

```diff
@@ -0,0 +1,140 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+import TableCell from '@material-ui/core/TableCell';
+import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import isEmpty from 'lodash/isEmpty';
+import { PREVIEW_STATUS } from 'services/PreviewStatus';
+import Heading, { HeadingTypes } from 'components/Heading';
+import ThemeWrapper from 'components/ThemeWrapper';
+
+const CustomTableCell = withStyles((theme) => ({
+  head: {
+    backgroundColor: theme.palette.grey['300'],
+    color: theme.palette.common.white,
+    padding: 10,
+    fontSize: 14,
+  },
+  body: {
+    padding: 10,
+    fontSize: 14,
+  },
+}))(TableCell);
+
+export const messageTextStyle = {
+  fontSize: '1.3rem !important',
+  margin: '10px 0',
+};
+const styles = (theme) => ({
+  root: {
+    width: '100%',
+    display: 'inline-block',
+    height: 'auto',
+    marginTop: theme.spacing(1),
+  },
+  table: {
+    width: '100%',
+  },
+  row: {
+    height: 40,
+    '&:nth-of-type(odd)': {
+      backgroundColor: theme.palette.grey['600'],
+    },
+  },
+  messageText: messageTextStyle,
+});
+
+interface IDataTableProps extends WithStyles<typeof styles> {
+  headers: string[];
+  records: any[];
+  isInput?: boolean;
+  previewStatus?: string;
+}
+
+const DataTableView: React.FC<IDataTableProps> = ({
+  classes,
+  headers,
+  records,
+  isInput,
+  previewStatus,
+}) => {
+  const getStatusMsg = () => {
+    let msg;
+    const recordType = isInput ? 'Input' : 'Output';
+    if (previewStatus === PREVIEW_STATUS.RUNNING) {
+      // preview is still running but there's no data yet
+      msg = `${recordType} records have not been generated yet. Please check again in a few minutes.`;
+    } else if (previewStatus !== PREVIEW_STATUS.RUNNING) {
+      // not running preview but there is no preview data
+      msg = `${recordType} records have not been generated. Please verify your logic or try sending more data.`;
+    } else {
+      // default message
+      msg = `${recordType} records are not available. Please try running preview again.`;
+    }
+    return msg;
+  };
+
+  if (isEmpty(records)) {
+    return (
+      <div>
+        <Heading type={HeadingTypes.h3} label={getStatusMsg()} className={classes.messageText} />
+      </div>
+    );
+  }
+
+  return (
+    <Paper className={classes.root}>
+      <Table>
+        <TableHead>
+          <TableRow className={classes.row}>
+            {headers.map((fieldName) => {
+              return <CustomTableCell>{fieldName}</CustomTableCell>;
```

Comment:
Since we are using `map`, I believe we should assign `key` to each element.

Reply:
done!